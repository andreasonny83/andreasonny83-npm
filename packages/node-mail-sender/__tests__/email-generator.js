// Copyright (c) 2018 AndreaSonny <andreasonny83@gmail.com> (https://github.com/andreasonny83)
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

const { EmailGenerator } = require('../src/email-generator');

describe('EmailGenerator', () => {
  const emailTemplateMock = (from, sender, date, body, footer) =>
    `Hello Jest,
    On ${date}, ${from} <${sender}> left you the following message:
    ${body}
    ${footer}`;

  beforeEach(() => (emailGenerator = new EmailGenerator(emailTemplateMock)));

  it('should export a function', () => {
    // Assert
    expect(typeof EmailGenerator).toEqual('function');
  });

  it('should set the charset to `UTF-8` by default', () => {
    // Arrange
    const expectedCharset = 'UTF-8';

    // Act
    const result = emailGenerator.generate();

    // Assert
    expect(result.Message.Subject.Charset).toEqual(expectedCharset);
  });

  it('should accept a different charset', () => {
    // Arrange
    const expectedCharset = 'ISO-8859-1';

    // Act
    const isoEmailGenerator = new EmailGenerator(
      emailTemplateMock,
      expectedCharset
    );
    const result = isoEmailGenerator.generate();

    // Assert
    expect(result.Message.Subject.Charset).toEqual(expectedCharset);
  });

  it('should generate an email object', () => {
    // Arrange
    const emailGenerator = new EmailGenerator(emailTemplateMock);
    const subject = 'test email';
    const from = 'fake_user@mail.com';
    const sender = 'Heisenberg';
    const email = 'heisenberg@crystal.blue';
    const to = 'fake_recipient@mail.com';
    const body = 'Hello! This is the end of an era';

    // Mock the Date constructor
    Date = jest.fn(() => ({
      toUTCString: () => '31st December 1999',
    }));

    // Act
    const prepareMail = emailGenerator.generate(
      subject,
      body,
      from,
      to,
      sender,
      email
    );

    // Assert
    expect(prepareMail).toMatchSnapshot();
  });
});
