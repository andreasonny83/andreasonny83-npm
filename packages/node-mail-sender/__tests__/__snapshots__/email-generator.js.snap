// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailGenerator should generate an email object 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "fake_recipient@mail.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "Hello Jest,
    On 31st December 1999, heisenberg@crystal.blue <Heisenberg> left you the following message:
    Hello! This is the end of an era
    
    This is an automatic e-mail.
    If you do not wish to continue receiving these messages or for other queries, plase contact your web administrator.",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
      From: heisenberg@crystal.blue <Heisenberg> <br/>
      Date: 31st December 1999 <br/>
      Message: Hello! This is the end of an era <br/>
      Footer: 
    This is an automatic e-mail.
    If you do not wish to continue receiving these messages or for other queries, plase contact your web administrator. <br/>
      ",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "test email",
    },
  },
  "Source": "fake_user@mail.com",
}
`;
